dace/codegen/targets/cuda.py

```diff
@@ -1002,12 +1094,43 @@ def generate_scope(self, sdfg, dfg_scope, state_id, function_stream,
                                        state.node_id(scope_entry))
 
         # Comprehend grid/block dimensions from scopes
-        grid_dims, block_dims, tbmap, dtbmap = self.get_kernel_dimensions(
-            dfg_scope)
+        grid_dims, block_dims, tbmap, dtbmap =\
+            self.get_kernel_dimensions(dfg_scope)
+        is_persistent = (dfg_scope.source_nodes()[0].map.schedule
+                         == dtypes.ScheduleType.GPU_Persistent)
 
         # Get parameters of subgraph
         kernel_args = dfg_scope.arglist()
+
+        # handle dynamic map inputs
+        # TODO move this loop into dfg_scope.arglist()
+        for e in dace.sdfg.dynamic_map_inputs(state, scope_entry):
+            kernel_args[str(e.src)] = e.src.desc(sdfg)
```

Comment:
Do we still need these lines?

Reply:
If there was a fix for dynamic range maps on GPU in main repo then no, otherwise this is the fix. Though I can remove the TODO.

Reply:
I was going to ask "what happens if you print the arglist", if the dynamic range is there we can remove those loops from the code.